source/config.cpp

```diff
@@ -84,115 +84,107 @@ void Config::read(string const &file_name)
 	string line;
 
 	while( std::getline(f, line) ) {
-		if( line.size() ) {
-			if( line[0] == '#' ) continue;
+		if( !line.size() or line[0] == '#' ) continue;
 
-			if( line.back() == '\r' ) {
-				line.pop_back();
-				if( line.empty() ) continue;
+		if( line.back() == '\r' ) {
+			line.pop_back();
+			if( line.empty() ) continue;
+		}
+
+		if( !(line[0] == '+' or line[0] == '-') ) throw std::runtime_error("Invalid token at the begining of line in config file! Each line should begin with ether '+' or '-' or '#'! Line: " + line);
```

Comment:
`if( line[0] != '+' and line[0] != '-') ...`

Reply:
ok, done